src/contexts/account/useProposalVotes.ts

```diff
@@ -3,14 +3,116 @@ import { useCallback } from 'react';
 import useAddresses from 'src/hooks/useAddresses';
 import { VoteType } from 'src/types';
 import celoRegistry from 'src/utils/celoRegistry';
-import { useAccount, usePublicClient } from 'wagmi';
+import { Address, PublicClient, useAccount, usePublicClient } from 'wagmi';
 
 interface Vote {
   vote: VoteType;
   weight: string;
 }
+interface VoteEventArgs {
+  voter?: Address | undefined;
+  proposalId?: bigint | undefined;
+  yesVotes?: bigint | undefined;
+  noVotes?: bigint | undefined;
+  abstainVotes?: bigint | undefined;
+}
 type ProposalID = string;
 
+function extractVote(args: VoteEventArgs): 'yesVotes' | 'noVotes' | 'abstainVotes' | undefined {
```

Comment:
seems potentially risky.

I feel like there should be a different return if the vote is 0 vs if it is not present. right now both will return undefined

Reply:
It's the same thing though, not voting or revoking votes results in the same UI

Reply:
ok so cases here, 

if vote args has a key with value that is anything but `0n` return that key as the Vote type. 


i thought only a vote Event where all keys were zero would be a revoke vote. 

if there is not a chance that the vote event could say return some different values then i see how it will be fine. 

last q  *instead of `undefined` should we return `revokeVote` ? *

Reply:
Your assumptions are correct regarding the event vote.

Regarding the `undefined` vs `revokedVote`, it's the reason I replied so late (outside having forgotten about this PR), it'd require a fairly big refactor